apps/omg/lib/omg/state/transaction/fee_token_claim.ex

```diff
@@ -114,14 +136,28 @@ defimpl OMG.State.Transaction.Protocol, for: OMG.State.Transaction.FeeTokenClaim
   def get_inputs(%Transaction.FeeTokenClaim{}), do: []
 
   @doc """
-  Fee claiming transaction is not used to transfer funds
+  Tells whether Fee claiming transaction is valid
   """
   @spec valid?(Transaction.FeeTokenClaim.t(), Transaction.Signed.t()) :: {:error, atom()}
-  def valid?(%Transaction.FeeTokenClaim{}, _signed_tx), do: {:error, :not_implemented}
+  def valid?(%Transaction.FeeTokenClaim{} = fee_tx, _signed_tx) do
+    # we're able to check structure validity => single output with amount > 0
+    with outputs = Transaction.get_outputs(fee_tx),
+         true <- length(outputs) == 1 || {:error, :wrong_number_of_fee_outputs},
+         [output] = outputs,
+         true <- output.amount > 0 || {:error, :fee_output_amount_has_to_be_positive},
+         do: true
+  end
 
   @doc """
   Fee claiming transaction is not used to transfer funds
   """
-  @spec can_apply?(Transaction.FeeTokenClaim.t(), list(Output.t())) :: {:error, atom()}
-  def can_apply?(%Transaction.FeeTokenClaim{}, _outputs_spent), do: {:error, :not_implemented}
+  @spec can_apply?(Transaction.FeeTokenClaim.t(), list(Output.t())) :: {:ok, map()} | {:error, atom()}
+  def can_apply?(%Transaction.FeeTokenClaim{outputs: [claimed]}, outputs) do
+    with %Output{} = collected <-
+           Enum.find(outputs, {:error, :surplus_in_token_not_collected}, fn o -> o.currency == claimed.currency end),
+         true <- collected.amount == claimed.amount || {:error, :claimed_and_collected_amounts_mismatch},
```

Comment:
question: is this the `error` that will pop up in the byzantine event of operator attempting to drain too much of fees in a block?

Reply:
So there could be to cases:
* `{:error, :surplus_in_token_not_collected}`  - fee tx claiming the token that state knows nothing about,
* `{:error, :claimed_and_collected_amounts_mismatch}` - collected and claimed aren't equal (we have to claim exactly the amount of collected)